middleware/admin/certificate_v1.py

```diff
@@ -214,11 +220,9 @@ def save_to_jsonfile(self, path):
             file.write("%s\n" % json.dumps(self.to_dict(), indent=2))
 
     def _parse(self, certificate_map):
-        if "version" not in certificate_map or certificate_map["version"] != self.VERSION:
-            raise ValueError(
-                "Invalid or unsupported HSM certificate version "
-                f"(current version is {self.VERSION})"
-            )
+        if certificate_map.get("version") != self.VERSION:
+            raise ValueError("Invalid or unexpected HSM certificate version "
```

Comment:
Same as previous comment, maybe also print the version we received

Reply:
Done